### PR TITLE
Get all favorites endpoint /favorites

### LINKED ISSUE #7: User can get all favorited songs with /favorites endpoint
When the user sends a `GET api/v1/favorites` request

User can expect a successful response body as follows with a `200 status`:
```
[
  {
    "id": 1,
    "title": "We Will Rock You",
    "artistName": "Queen"
    "genre": "Rock",
    "rating": 88
  },
  {
    "id": 2,
    "title": "Careless Whisper",
    "artistName": "George Michael"
    "genre": "Pop",
    "rating": 93
  },
]
```
Return `200 status` with an empty array if there are no favorites in the database

Acceptance criteria:
- [x]  All tests passing
- [x]  Happy path
- [ ]  Sad path
- [ ]  Peer review
- [ ]  Deployed to Heroku

### PR DESCRIPTION
**Pull request by zacisaacson**
## Type of change 

- [x] :beetle: Bug fix (non-breaking change which fixes an issue)
- [x] :hatching_chick: New feature (non-breaking change which adds functionality)
- [x] :page_with_curl: This change requires a documentation update

## Summary of changes 

- Add happy path test for endpoint
- Complete functionality
- Create helper function removing created_at and update_at

## How Has This Been Tested?

- [x] :black_square_button: Integration testing 
- [ ] :white_square_button: Unit testing 

## Checklist:

- [x] :white_check_mark: I have performed a self-review of my own code
- [x] :eight_spoked_asterisk: I have commented my code, particularly in hard-to-understand areas
- [ ] :page_facing_up: I have made corresponding changes to the documentation
- [x] :no_entry_sign: My changes generate no new warnings
- [x] :heavy_check_mark: I have added tests that prove my fix is effective or that my feature works
- [x] :100: New and existing unit tests pass locally with my changes

### CODE REVIEW

File: tests/getAllFavorites.spec.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+// describe('Test the sad path for get favorites endpoint', () => {
+//   it('will return 204 if no favorites', async () => {
+//        const res = await request(app)
+//          .get("/api/v1/favorites")
+//          expect(res.statusCode).toEqual(204);
+//          expect(res.body).toHaveProperty("error");
+//          expect(res.body.error).toBe("No content");
+//      });
+// });
```
Comment (zacisaacson) [inline tests/getAllFavorites.spec.js]: Still working through sad path tests, wasn't able to get this working but please let me know if you see anything I'm missing

File: helpers/favorites.js [hunk 2]
```diff
@@ -0,0 +1,8 @@
+function formatFavorites(favorites) {
+  favorites.map(favorite => {
+    delete favorite.created_at;
+    delete favorite.updated_at;
+    return favorite;
+  })
+};
+ module.exports = formatFavorites;
```
Comment (zacisaacson) [inline helpers/favorites.js]: Decided to create a helper function, let me know what you think

File: routes/api/v1/favorites.js [hunk 3]
```diff
@@ -41,4 +42,17 @@ router.post('/', (request, response) => {
       });
 });
 
+router.get('/', (request, response) => {
+  database('favorites').select()
```
Comment (lrs8810) [inline routes/api/v1/favorites.js]: @zacisaacson  did you try selecting the columns in the Knex query? That worked for a single resource and helped me avoid creating a helper method
```suggestion
  database('favorites').select('id', 'title', 'artistName', 'genre', 'rating')
```

File: tests/getAllFavorites.spec.js [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+  afterEach(() => {
+    database.raw('truncate table favorites cascade');
+  });
+
+  describe('GET /api/v1/favorites', () => {
+   it('happy path', async () => {
+     const res = await request(app)
+       .get("/api/v1/favorites")
+
+       expect(res.statusCode).toEqual(200);
```
Comment (lrs8810) [inline tests/getAllFavorites.spec.js]: Love the thorough testing!

File: tests/getAllFavorites.spec.js (hunk 1 above)
Comment (lrs8810) [reply tests/getAllFavorites.spec.js]: Sorry, this is supposed to be a 200 response with an empty array. I will update the card now.

File: routes/api/v1/favorites.js [hunk 5]
```diff
@@ -41,6 +41,13 @@ router.post('/', (request, response) => {
       });
 });
 
+router.get('/', (request, response) => {
+  database('favorites').select("id", "title", "artistName", "genre", "rating")
```
Comment (lrs8810) [inline routes/api/v1/favorites.js]: Thank you for making this change! It really cleans it up.

**Pull request merged**